src/mds/MDSDaemon.cc

```diff
@@ -1101,6 +1099,7 @@ void MDSDaemon::suicide()
     if (objecter->initialized.read()) {
       objecter->shutdown();
     }
+    timer.shutdown();
```

Comment:
In the MDSRank case, we shutdown timer before objecter.  However, looking at the things we call back from the timer, I don't think any of them do IO, so this should be safe too.